File: Codility/L1-BinaryGap.cpp
int solution(int N) {
    // write your code in C++14 (g++ 6.2.0)
    int maxcnt = 0, cnt = 0;
    bool rflag = false;
    
    while (N > 0) {
        if (N & 0x1) {
            if (rflag == false) {
                rflag = true;
            }
            else {
                maxcnt = max(maxcnt, cnt);
            }
            
            cnt = 0;
        } else {
            ++ cnt;
        }
        
        N /= 2;
    }
    
    return maxcnt;
}

File: Codility/L1-BinaryGap.py
def solution(N):
    # write your code in Python 3.6
    right = -1
    cur = 0
    maxlen = 0
    
    while (N):
        if (N & 1):
            if (right >= 0):
                maxlen = max(cur - right - 1, maxlen)
            right = cur
        N >>= 1
        cur += 1
        
    return maxlen
